test/test_enzyme.jl

```diff
@@ -23,3 +23,86 @@ f(grid) = CenterField(grid)
 
     @test size(primal) == size(shadow)
 end
+
+@testset "Test Constructor Any Bug" begin
+    using Oceananigans.Fields: FunctionField
+    using Oceananigans: architecture
+    using KernelAbstractions
+
+    Enzyme.API.runtimeActivity!(true)
+    Enzyme.API.looseTypeAnalysis!(true)
+    Enzyme.EnzymeRules.inactive_type(::Type{<:Oceananigans.Grids.AbstractGrid}) = true
+    Enzyme.EnzymeRules.inactive_type(::Type{<:Oceananigans.Clock}) = true
+    Enzyme.EnzymeRules.inactive_noinl(::typeof(Core._compute_sparams), args...) = nothing
+
+    Nx = Ny = 64
+    Nz = 8
+
+    x = y = (-π, π)
+    z = (-0.5, 0.5)
+    topology = (Periodic, Periodic, Bounded)
+
+    grid = RectilinearGrid(size=(Nx, Ny, Nz); x, y, z, topology)
+
+    model = HydrostaticFreeSurfaceModel(; grid,
+                                        tracers = :c,
+                                        buoyancy = nothing)
+
+    function set_initial_condition!(model_tracer, amplitude)
+        # Set initial condition
+        amplitude = Ref(amplitude)
+
+        # This has a "width" of 0.1
+        cᵢ(x, y, z) = amplitude[]
+        temp = Base.broadcasted(Base.identity, FunctionField((Center, Center, Center), cᵢ, model_tracer.grid))
+
+        temp = convert(Base.Broadcast.Broadcasted{Nothing}, temp)
+        grid = model_tracer.grid
+        arch = architecture(model_tracer)
+        bc′ = temp
+
+        param = Oceananigans.Utils.KernelParameters(size(model_tracer), map(offset_index, model_tracer.indices))
+        Oceananigans.Utils.launch!(arch, grid, param, _broadcast_kernel!, model_tracer, bc′)
+
+        return nothing
+    end
```

Comment:
I think this and subsequent functions might be best moved out of the test set scope